### PR TITLE
Adding support to change serializer and digest.

### PR DESCRIPTION
**Pull request by grumpy**
Adding support to change serializer and digest during rotation.  Added safe json serializer for rails 3.

### CODE REVIEW

File: lib/safe_json_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+      flash.update(json.delete('flash').to_h.with_indifferent_access.symbolize_keys)
+      json['flash'] = flash
+    end
+    json
+  end
+
+  module_function :load
+
+  def dump(obj, anIO = nil, limit = nil)
+    obj.to_json
```
Comment (noku) [inline lib/safe_json_serializer.rb]: maybe use: 

```ruby
JSON.generate(obj, quirks_mode: true)
```

similar to what was used in rails: [json_serializer](https://github.com/rails/rails/pull/13692/files#diff-cf6d7cf861425a8fad71dbbbeb70f739R9)

File: lib/safe_json_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module SafeJSONSerializer
```
Comment (noku) [inline lib/safe_json_serializer.rb]: why `module` instead of `class`? using `class` will remove additional `module_function`

**Review by noku — APPROVED** [review]

File: lib/safe_json_serializer.rb (hunk 2 above)
Comment (grumpy) [reply lib/safe_json_serializer.rb]: I was imitating what is implemented for the other serializers.  I can try a class.

File: lib/safe_json_serializer.rb (hunk 1 above)
Comment (grumpy) [reply lib/safe_json_serializer.rb]: Done

**Review by dehzhas — COMMENTED** [review]
I don't really feel I'm in a position to give a terribly helpful look at this. I don't see anything strange here, but I'm not really sure how it fits in the overall serializer and digest for Rails.

**Pull request merged**